src/Console/Command/ExecuteCommand.php

```
@@ -0,0 +1,119 @@
+
+        // format of execute is "vendor/package:version@binary"
+        /** @var string[] $deps */
+        $deps = $input->getOption('deps') ?? [];
+        $name = $input->getArgument('package');
+        $nameSplitted = explode('@', $name);
+
+        if (\count($nameSplitted) >= 2) {
+            $deps[] = $nameSplitted[0];
+            $binary = $nameSplitted[1];
+        } else {
+            $deps[] = $name;
+            $binary = null;
+        }
+
+        $fs = new Filesystem();
+        $composerJsonPath = $tmpDir . '/composer.json';
+        $vendorDirectory = $tmpDir . '/vendor';
+        $binaryDirectory = $tmpDir . '/bin';
+        $tokens = $this->getRawTokens($input);
+
+        $args = [];
+        $foundPackageName = false;
+
+        foreach ($tokens as $token) {
+            if ($foundPackageName) {
+                $args[] = $token;
+            } else {
+                if ($token === $name) {
+                    $foundPackageName = true;
+                }
+            }
+        }
+
+        try {
+            $fs->mkdir($tmpDir, 0o755);
+            /** @var list<string> $composerArgs */
+            $composerArgs = ['require', ...$deps, '--no-interaction'];
+            $this->composer->run($composerJsonPath, $vendorDirectory, $composerArgs, $output, $input->isInteractive(), $binaryDirectory);
+
+            if (null === $binary) {
+                // / Get first binary declared in the package if none was specified
+                $lockFilePath = $tmpDir . '/composer.lock';
+                $lockContent = json_decode(file_get_contents($lockFilePath) ?: '{}', true);
+
+                $foundPackage = [];
+
+                foreach ($lockContent['packages'] as $package) {
+                    if ($package['name'] === $nameSplitted[0]) {
+                        $foundPackage = $package;
+
+                        break;
+                    }
+                }
+
+                $binary = $foundPackage['bin'][0] ?? null;
```

should we raise an exception in case there is no binary?